lib/storage/search_options.go

```diff
@@ -0,0 +1,53 @@
+package storage
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/VictoriaMetrics/metrics"
+)
+
+// searchOptions contains common options for Search.
+type searchOptions struct {
+	deadline uint64
+
+	readMetricIDs *atomic.Uint64
+	source        string
+}
+
+// getSearchOptions returns new searchOptions.
+// Note that the readMetricIDs metric is only flushed when the searchOptions is put back with putSearchOptions.
+func getSearchOptions(deadline uint64, source string) *searchOptions {
+	if v := searchOptionsPool.Get(); v != nil {
+		so := v.(*searchOptions)
+		so.deadline = deadline
+		so.source = source
```

Comment:
I'd prefer to have `so.readMetricIDs.Store(0)` right here, so it is clear that reset happens in one place